### PR TITLE
Add return status code support for plugin functions

### PR DESCRIPTION
**Pull request by sod-lol**
Add return status code support for plugin functions
related to #687

### CODE REVIEW

**Review by avelino — CHANGES_REQUESTED** [review]
Need update plugin doc page 

- https://docs.prestd.com/prestd/api-reference/plugins/
- https://github.com/prest/prest/blob/main/docs/api-reference/plugins.md

Review tests https://github.com/prest/prest/actions/runs/3603785620/jobs/6074271370#step:10:51

**Review by arxdsilva — APPROVED** [review]

**Comment by arxdsilva** [comment]
Please fix tests (:

File: plugins/plugins.go [hunk 1]
```diff
@@ -61,9 +67,25 @@ func loadFunc(fileName, funcName string, r *http.Request) (ret string, err error
...
-	log.Println("ret plugin:", ret)
+	// Exec (call) function name, return string (In case which return status code does not matter)
+	function, ok := f.(func() string)
+
+	if !ok {
+		// It is probable that plugin function return not only json but also status code.
+		function := f.(func() (string, int))
+		retJson, code := function()
+		ret.ReturnJson = retJson
+		ret.StatusCode = code
```
Comment (avelino) [inline plugins/plugins.go:78]: if `code` is `nil`, set default value `http.StatusOK`

when you don't set the `status code` I think it's better to have a default return of `200` (aka `http.StatusOK`)

does this proposal make sense @sod-lol @arxdsilva?

**Comment by sod-lol** [comment]
Hi guys. I recently faced some personal problems. i will try to fix issue once i solved my problems.

File: plugins/plugins.go (hunk 1 above)
Comment (sod-lol) [reply plugins/plugins.go:78]: Hi.
The type of code variable is int, so it cannot be nil unless we change the type of code variable, but don't you agree that forcing a developer to explicitly return status code minimizes the error? What is your opinion? @avelino @arxdsilva

**Review by avelino — APPROVED** [review]

**Pull request merged**